backend/capellacollab/sessions/routes.py

```diff
@@ -87,70 +100,72 @@ def get_current_sessions(
 
 
 @router.post(
-    "/",
+    "/readonly",
     response_model=AdvancedSessionResponse,
 )
 def request_session(
-    body: PostSessionRequest,
+    body: PostReadonlySessionRequest,
     db_user: DatabaseUser = Depends(get_own_user),
     operator: Operator = Depends(get_operator),
     db: Session = Depends(get_db),
     token=Depends(JWTBearer()),
 ):
-    assert body.type == WorkspaceType.READONLY
-
     rdp_password = generate_password(length=64)
 
-    owner = db_user.username
+    owner = db_user.name
 
     log.info("Starting persistent session creation for user %s", owner)
 
-    existing_user_sessions = database.get_sessions_for_user(db, owner)
+    project = get_project_by_slug(db, body.project_slug)
+    if not project:
+        raise HTTPException(
+            status.HTTP_404_NOT_FOUND,
+            {
+                "reason": f"The project with name {body.project_slug} was not found.",
+                "technical": f"No project {body.project_slug} found.",
+            },
+        )
 
-    if body.repository in [
-        session.repository for session in existing_user_sessions
-    ]:
+    verify_project_role(project.name, token, db)
+
+    model = get_model_by_slug(db, body.project_slug, body.model_slug)
+    if not model:
         raise HTTPException(
-            status_code=404,
-            detail={
-                "err_code": "existing_session",
-                "reason": f"You already have a open Read-Only Session for the repository {body.repository}. Please navigate to 'Active Sessions' to Reconnect",
+            status.HTTP_404_NOT_FOUND,
+            {
+                "reason": f"The model with name {body.model_slug} was not found.",
+                "technical": f"No model {body.model_slug} found.",
             },
         )
```

Comment:
How is this resolved? There is no comment and I don't see it applied.

Reply:
Yes it is: the comment is marked "outdated"

Reply:
It shows as outdated, but I don't see the suggestion applied. Am I missing something?

Reply:
![image](https://user-images.githubusercontent.com/23395732/201302829-0632e47b-569c-4876-9fa7-ed89b652ad88.png)

Reply:
> Same here. Please use the `injectables` instead: `get_existing_capella_model`

This does not apply, since the `model_slug` is not part of the URL. I can reuse that function though.